realm/src/androidTest/java/io/realm/NotificationsTest.java

```diff
@@ -94,6 +94,70 @@ public Boolean call() throws Exception {
         assertTrue(Realm.realmsCache.get().isEmpty());
     }
 
+    public void testRemoveNotifications () throws InterruptedException, ExecutionException {
```

Comment:
This test case doesn't have to be this complicated. Something like this will also work:

```
        final AtomicInteger counter= new AtomicInteger(0);
        RealmChangeListener listener = new RealmChangeListener() {
            @Override
            public void onChange() {
                counter.incrementAndGet();
            }
        };

        realm = Realm.getInstance(getContext());
        realm.addChangeListener(listener);
        realm.removeChangeListener(listener);

        realm.beginTransaction();
        realm.createObject(AllTypes.class);
        realm.commitTransaction();

        assertEquals(0, counter.get());
```